plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -158,16 +123,13 @@ export class TeamManager {
         } else {
             // TODO: #7422 Temporary conditional to test experimental feature
             if (this.experimentalLastSeenAtEnabled) {
-                const eventCacheKey = JSON.stringify([team.id, event])
-                const timestamp = DateTime.now().valueOf()
-                if ((this.eventLastSeenCache.get(eventCacheKey) ?? 0) < timestamp) {
-                    this.eventLastSeenCache.set(eventCacheKey, timestamp)
-                }
-                // TODO: Allow configuring this via env vars
-                // We flush here every 2 mins (as a failsafe) because the main thread flushes every minute
-                if (this.eventLastSeenCache.size > 1000 || DateTime.now().diff(this.lastFlushAt).as('seconds') > 120) {
-                    // to not run out of memory
-                    await this.flushLastSeenAtCache()
+                if ((this.eventLastSeenCache.get(cacheKey) ?? 0) < cacheTime) {
+                    this.eventLastSeenCache.set(cacheKey, cacheTime)
+                    await this.db.postgresQuery(
+                        `UPDATE posthog_eventdefinition SET last_seen_at=$1 WHERE team_id=$2 and name=$3`,
+                        [DateTime.now(), team.id, event],
```

Comment:
I think we should be consistent in the DB here too and avoid future confusion. Let's change `last_seen_at` to a `DateField` and only store the current date. Should also make the queries marginally faster.

Reply:
I'm very afraid of doing this. It'll lock the table for a good minute or more (worse case? possibly on cloud?), which will just cause chaos and a broken deploy. We're stuck with this now. It's fine. :)

Reply:
Well we could create a separate field, migrate the data and destroy the old one. Changing the name even makes sense as the at suffix suggests a date time. But we can do this later.